File: packages/core/tests/fixtures/unhealthy/Complex.java
public class Complex {
    public String processData(
        Object[] items, Object config, Object options,
        Object callback, String mode, Object extra
    ) {
        if (items != null) {
            for (Object item : items) {
                int val = (int) item;
                if (val > 0) {
                    while (val > 1) {
                        if (val % 2 == 0) {
                            if ("fast".equals(mode)) {
                                return "fast";
                            } else if ("slow".equals(mode)) {
                                return "slow";
                            }
                        }
                        val--;
                    }
                } else if (val < -10) {
                    return "negative";
                }
            }
        } else if (extra != null) {
            return "extra";
        }
        return "default";
    }
}

File: packages/core/tests/fixtures/holdout/unhealthy/complex.py
"""Heavily nested and complex function — unhealthy health score expected."""


def process_data(items, config, options, callback, mode, extra):
    if items:
        for item in items:
            if item > 0:
                while item > 1:
                    if item % 2 == 0:
                        if mode == 'fast':
                            if config.get('turbo'):
                                if options.get('aggressive'):
                                    return 'turbo-fast'
                                return 'fast-config'
                            return 'fast'
                        elif mode == 'slow':
                            if config.get('safe'):
                                return 'slow-safe'
                            return 'slow'
                        elif mode == 'auto':
                            if item > 100:
                                return 'auto-large'
                            return 'auto'
                    item -= 1
            elif item < -10:
                return 'negative'
            elif item == 0:
                return 'zero'
    elif extra:
        if isinstance(extra, dict):
            if 'key' in extra:
                return extra['key']
        return 'extra'
    return 'default'
